### ISSUE TITLE
[Bug] Bug when transcoding 4K content but ok when it's from less resolutions.

### ISSUE THREAD

**Issue by CyberPoison** [linuxserver/docker-jellyfin#21]
## How to reproduce 
1. Go to PlayBack 
2. Set Video Accelartion API (VAP-API)
2.1 set /dev/dri/renderD128 on the render node
2. Enable HW Acceleration.
3. Play some 4K HEVC videos and try to transcode it.

------------------------------

## Expected Behavior
Transcode 4K content to h264 as expected.

## Current Behavior
"PlaybackErrorNoCompatibleStream"

## Environment
**OS: Linux
**CPU architecture:** x64     
**How docker service was installed:**      

## Command used to create docker container (run/create/compose/screenshot)
```
docker create \
  --name=jellyfin \
  -e PUID=0 \
  -e PGID=1000 \
  -e TZ=Europe/London \
  -e UMASK_SET=022 \
  -p 8096:8096 \
  -p 8920:8920 \
  -v /opt/appdata/jellyfin:/config \
  -v /mnt:/mnt \
  -v /opt/transcodes:/transcode `#optional` \
  -v /opt/vc/lib:/opt/vc/lib \
  --device /dev/dri:/dev/dri \
  --device /dev/vchiq:/dev/vchiq \
  --restart unless-stopped \
  linuxserver/jellyfin
```
## ffmpeg logs
```
/usr/lib/jellyfin-ffmpeg/ffmpeg -hwaccel vaapi -hwaccel_output_format vaapi -vaapi_device /dev/dri/renderD128 -i file:"/mnt/tcrypt/EN/Movies4K[EN]/Furious 7/Furious 7.2015.Extended Cut.UHD.BluRay.Remux.2160p.mkv" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 h264_vaapi  -b:v 272000 -maxrate 272000 -bufsize 544000 -profile:v high -level 41 -force_key_frames:0 "expr:gte(t,0+n_forced*3)" -vf "format=nv12|vaapi,hwupload,scale_vaapi=w=426:h=238" -copyts -vsync -1 -codec:a:0 aac -strict experimental -ac 2 -ab 128000 -af "volume=2" -f hls -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -hls_time 3 -individual_header_trailer 0 -hls_segment_type mpegts -start_number 0 -hls_segment_filename "/config/data/transcoding-temp/transcodes/34c33d415643cd1f181a9527b8dbdb31%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/config/data/transcoding-temp/transcodes/34c33d415643cd1f181a9527b8dbdb31.m3u8"


ffmpeg version 4.2.1 Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.4.0-1ubuntu1~18.04.1)
  configuration: --toolchain=hardened --prefix=/usr --target-os=linux --enable-cross-compile --extra-cflags=--static --enable-gpl --enable-static --disable-doc --disable-ffplay --disable-shared --disable-libxcb --disable-sdl2 --disable-xlib --enable-libfontconfig --enable-fontconfig --enable-gmp --enable-gnutls --enable-libass --enable-libbluray --enable-libdrm --enable-libfreetype --enable-libfribidi --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libwebp --enable-libx264 --enable-libx265 --enable-libzvbi --enable-omx --enable-omx-rpi --enable-version3 --enable-vaapi --enable-vdpau --arch=amd64 --enable-nvenc --enable-nvdec --enable-amf
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
Input #0, matroska,webm, from 'file:/mnt/tcrypt/EN/Movies4K[EN]/Furious 7/Furious 7.2015.Extended Cut.UHD.BluRay.Remux.2160p.mkv':
  Metadata:
    encoder         : libebml v1.3.6 + libmatroska v1.4.9
    creation_time   : 2018-08-26T11:20:53.000000Z
  Duration: 02:19:54.88, start: 0.000000, bitrate: 70142 kb/s
    Chapter #0:0: start 0.000000, end 279.946000
    Metadata:
      title           : 00:00:00.000
    Chapter #0:1: start 279.946000, end 689.480000
    Metadata:
      title           : 00:04:39.946
    Chapter #0:2: start 689.480000, end 999.290000
    Metadata:
      title           : 00:11:29.480
    Chapter #0:3: start 999.290000, end 1554.052000
    Metadata:
      title           : 00:16:39.290
    Chapter #0:4: start 1554.052000, end 2023.897000
    Metadata:
      title           : 00:25:54.052
    Chapter #0:5: start 2023.897000, end 2504.585000
    Metadata:
      title           : 00:33:43.897
    Chapter #0:6: start 2504.585000, end 2801.006000
    Metadata:
      title           : 00:41:44.585
    Chapter #0:7: start 2801.006000, end 3130.711000
    Metadata:
      title           : 00:46:41.006
    Chapter #0:8: start 3130.711000, end 3751.414000
    Metadata:
      title           : 00:52:10.711
    Chapter #0:9: start 3751.414000, end 3976.681000
    Metadata:
      title           : 01:02:31.414
    Chapter #0:10: start 3976.681000, end 4456.869000
    Metadata:
      title           : 01:06:16.681
    Chapter #0:11: start 4456.869000, end 4954.032000
    Metadata:
      title           : 01:14:16.869
    Chapter #0:12: start 4954.032000, end 5601.137000
    Metadata:
      title           : 01:22:34.032
    Chapter #0:13: start 5601.137000, end 5907.485000
    Metadata:
      title           : 01:33:21.137
    Chapter #0:14: start 5907.485000, end 6340.084000
    Metadata:
      title           : 01:38:27.485
    Chapter #0:15: start 6340.084000, end 6834.452000
    Metadata:
      title           : 01:45:40.084
    Chapter #0:16: start 6834.452000, end 7111.062000
    Metadata:
      title           : 01:53:54.452
    Chapter #0:17: start 7111.062000, end 7579.864000
    Metadata:
      title           : 01:58:31.062
    Chapter #0:18: start 7579.864000, end 7891.175000
    Metadata:
      title           : 02:06:19.864
    Chapter #0:19: start 7891.175000, end 8394.880000
    Metadata:
      title           : 02:11:31.175
    Stream #0:0: Video: hevc (Main 10), yuv420p10le(tv, bt2020nc/bt2020/smpte2084), 3840x2160 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 1k tbn, 23.98 tbc (default)
    Metadata:
      BPS-eng         : 41954647
      DURATION-eng    : 02:19:53.969000000
      NUMBER_OF_FRAMES-eng: 201254
      NUMBER_OF_BYTES-eng: 44020751797
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:1(rus): Audio: ac3, 48000 Hz, 5.1(side), fltp, 448 kb/s (default)
    Metadata:
      title           : Dub, Blu-Ray
      BPS-eng         : 448000
      DURATION-eng    : 02:19:54.272000000
      NUMBER_OF_FRAMES-eng: 262321
      NUMBER_OF_BYTES-eng: 470079232
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:2(rus): Audio: dts (DTS-HD MA), 48000 Hz, 5.1(side), s32p (24 bit)
    Metadata:
      title           : Dub
      BPS-eng         : 4243204
      DURATION-eng    : 02:19:54.070000000
      NUMBER_OF_FRAMES-eng: 786944
      NUMBER_OF_BYTES-eng: 4452219232
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:3(rus): Audio: dts (DTS-HD MA), 48000 Hz, 7.1, s32p (24 bit)
    Metadata:
      title           : Ð›. Ð’Ð¾Ð»Ð¾Ð´Ð°Ñ€ÑÐºÐ¸Ð¹
      BPS-eng         : 4979860
      DURATION-eng    : 02:19:54.880000000
      NUMBER_OF_FRAMES-eng: 787020
      NUMBER_OF_BYTES-eng: 5225666152
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:4(rus): Audio: dts (DTS-HD MA), 48000 Hz, 7.1, s32p (24 bit)
    Metadata:
      title           : Ð®. Ð–Ð¸Ð²Ð¾Ð²
      BPS-eng         : 5097396
      DURATION-eng    : 02:19:54.859000000
      NUMBER_OF_FRAMES-eng: 787018
      NUMBER_OF_BYTES-eng: 5348990936
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:5(rus): Audio: dts (DTS-HD MA), 48000 Hz, 7.1, s32p (24 bit)
    Metadata:
      title           : Ð”. Ð•ÑÐ°Ñ€ÐµÐ²
      BPS-eng         : 4980736
      DURATION-eng    : 02:19:54.870000000
      NUMBER_OF_FRAMES-eng: 787019
      NUMBER_OF_BYTES-eng: 5226579208
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:6(rus): Audio: dts (DTS-HD MA), 48000 Hz, 7.1, s32p (24 bit)
    Metadata:
      title           : Ð. ÐœÐ°Ñ‚Ð²ÐµÐµÐ²
      BPS-eng         : 4973578
      DURATION-eng    : 02:19:54.880000000
      NUMBER_OF_FRAMES-eng: 787020
      NUMBER_OF_BYTES-eng: 5219074056
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:7(eng): Audio: dts (DTS-HD HRA), 48000 Hz, 7.1, fltp
    Metadata:
      BPS-eng         : 3455999
      DURATION-eng    : 02:19:54.059000000
      NUMBER_OF_FRAMES-eng: 786943
      NUMBER_OF_BYTES-eng: 3626233344
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:8(rus): Subtitle: subrip (default)
    Metadata:
      title           : forced
      BPS-eng         : 138
      DURATION-eng    : 00:00:38.529000000
      NUMBER_OF_FRAMES-eng: 11
      NUMBER_OF_BYTES-eng: 666
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:9(rus): Subtitle: subrip
    Metadata:
      title           : full
      BPS-eng         : 68
      DURATION-eng    : 02:19:01.024000000
      NUMBER_OF_FRAMES-eng: 1416
      NUMBER_OF_BYTES-eng: 71137
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:10(eng): Subtitle: subrip
    Metadata:
      title           : full
      BPS-eng         : 45
      DURATION-eng    : 02:10:36.988000000
      NUMBER_OF_FRAMES-eng: 1463
      NUMBER_OF_BYTES-eng: 44102
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:11(eng): Subtitle: subrip
    Metadata:
      title           : SDH
      BPS-eng         : 49
      DURATION-eng    : 02:10:39.324000000
      NUMBER_OF_FRAMES-eng: 1670
      NUMBER_OF_BYTES-eng: 48602
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:12(eng): Subtitle: subrip
    Metadata:
      title           : SDH-Colored
      BPS-eng         : 60
      DURATION-eng    : 02:10:39.324000000
      NUMBER_OF_FRAMES-eng: 1670
      NUMBER_OF_BYTES-eng: 59535
      _STATISTICS_WRITING_APP-eng: mkvmerge v25.0.0 ('Prog Noir') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2018-08-26 11:20:53
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:13: Video: mjpeg (Progressive), yuvj444p(pc, bt470bg/unknown/unknown), 398x500 [SAR 300:300 DAR 199:250], 90k tbr, 90k tbn, 90k tbc (attached pic)
    Metadata:
      filename        : poster.jpg
      mimetype        : image/jpeg
Only '-vf format=nv12|vaapi,hwupload,scale_vaapi=w=426:h=238' read, ignoring remaining -vf options: Use ',' to separate filters
Only '-af volume=2' read, ignoring remaining -af options: Use ',' to separate filters
Stream mapping:
  Stream #0:0 -> #0:0 (hevc (native) -> h264 (h264_vaapi))
  Stream #0:1 -> #0:1 (ac3 (native) -> aac (native))
Press [q] to stop, [?] for help
[h264_vaapi @ 0x562ffb77c2c0] No usable encoding profile found.
Error initializing output stream 0:0 -- Error while opening encoder for output stream #0:0 - maybe incorrect parameters such as bit_rate, rate, width or height
[aac @ 0x562ffb7a4f40] Qavg: 63699.516
[aac @ 0x562ffb7a4f40] 2 frames left in the queue on closing
Conversion failed!
```
## Docker Logs
```
[21:05:12] [INF] /usr/lib/jellyfin-ffmpeg/ffmpeg -hwaccel vaapi -hwaccel_output_format vaapi -vaapi_device /dev/dri/renderD128 -i file:"/mnt/tcrypt/EN/Movies4K[EN]/Furious 7/Furious 7.2015.Extended Cut.UHD.BluRay.Remux.2160p.mkv" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -map -0:s -codec:v:0 h264_vaapi  -b:v 272000 -maxrate 272000 -bufsize 544000 -profile:v high -level 41 -force_key_frames:0 "expr:gte(t,0+n_forced*3)" -vf "format=nv12|vaapi,hwupload,scale_vaapi=w=426:h=238" -copyts -vsync -1 -codec:a:0 aac -strict experimental -ac 2 -ab 128000 -af "volume=2" -f hls -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -hls_time 3 -individual_header_trailer 0 -hls_segment_type mpegts -start_number 0 -hls_segment_filename "/config/data/transcoding-temp/transcodes/e0dc924247802ffba89f44078f7d678f%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/config/data/transcoding-temp/transcodes/e0dc924247802ffba89f44078f7d678f.m3u8",
[21:05:13] [ERR] FFMpeg exited with code 1,
[21:05:14] [WRN] cannot serve /config/data/transcoding-temp/transcodes/e0dc924247802ffba89f44078f7d678f0.ts as transcoding quit before we got there,
[21:05:14] [ERR] Error processing request: Could not find file '/config/data/transcoding-temp/transcodes/e0dc924247802ffba89f44078f7d678f0.ts'.,
```
Not having this issue on plex. 
btw this is my vainfo

```
vainfo
error: can't connect to X server!
libva info: VA-API version 1.1.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_1_1
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.1 (libva 2.1.0)
vainfo: Driver version: Intel i965 driver for Intel(R) Coffee Lake - 2.1.0
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Simple            : VAEntrypointEncSlice
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSliceLP
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointEncSlice
      VAProfileH264Main               : VAEntrypointEncSliceLP
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointEncSlice
      VAProfileH264High               : VAEntrypointEncSliceLP
      VAProfileH264MultiviewHigh      : VAEntrypointVLD
      VAProfileH264MultiviewHigh      : VAEntrypointEncSlice
      VAProfileH264StereoHigh         : VAEntrypointVLD
      VAProfileH264StereoHigh         : VAEntrypointEncSlice
      VAProfileVC1Simple              : VAEntrypointVLD
      VAProfileVC1Main                : VAEntrypointVLD
      VAProfileVC1Advanced            : VAEntrypointVLD
      VAProfileNone                   : VAEntrypointVideoProc
      VAProfileJPEGBaseline           : VAEntrypointVLD
      VAProfileJPEGBaseline           : VAEntrypointEncPicture
      VAProfileVP8Version0_3          : VAEntrypointVLD
      VAProfileVP8Version0_3          : VAEntrypointEncSlice
      VAProfileHEVCMain               : VAEntrypointVLD
      VAProfileHEVCMain               : VAEntrypointEncSlice
      VAProfileHEVCMain10             : VAEntrypointVLD
      VAProfileHEVCMain10             : VAEntrypointEncSlice
      VAProfileVP9Profile0            : VAEntrypointVLD
      VAProfileVP9Profile0            : VAEntrypointEncSlice
      VAProfileVP9Profile2            : VAEntrypointVLD
```
on linuxserver/emby also it gives me that issue it does software transcoding instead of hw 

Kind Regards

**Comment by aptalca** [comment]
The error you're seeing is an encoding error. It doesn't even attempt to decode via hw. 

Compare that log to one when it works (lower res as you mentioned)

In any case, it's a question for jellyfin as we use the ffmpeg binary they provide

**Comment by CyberPoison** [comment]
@aptalca it will be fixed on the 10.5.0 (no date for this)  or please use the source of this docker jellyfin/jellyfin:nightly
It have enhancement and features improved
